### PR TITLE
1699

### PR DESCRIPTION
**Pull request by EllisHenderson**
Fixes Issue #647

### CODE REVIEW

**Comment by caroline-church** [comment]
@EllisHenderson can you link to the issue in the commit message on your pr please, rather than using zenhub to link to the pr. This is so we can see which issues have been closed in a release.

File: packages/composer-runtime/test/registry.js [hunk 1]
```diff
@@ -339,6 +339,17 @@ describe('Registry', () => {
                 .should.be.rejectedWith(AccessException);
         });
 
+        it('should throw if resource not specified', () => {
+
+            return registry.add({})
+                .should.be.rejectedWith(/Expected a Resource or Concept./);
+        });
+
+        it('should throw if resource not specified', () => {
```
Comment (bestbeforetoday) [inline packages/composer-runtime/test/registry.js]: This test should be in the _describe_ that tests `update`, not in the suite that tests `add`.

File: packages/composer-runtime/lib/registry.js [hunk 2]
```diff
@@ -228,13 +236,23 @@ class Registry extends EventEmitter {
...
-        let id = resource.getIdentifier();
-        let object = this.serializer.toJSON(resource, {
-            convertResourcesToRelationships: options.convertResourcesToRelationships
-        });
-        object = this.addInternalProperties(object);
-        return this.dataCollection.get(id)
+        let id;
+        let object;
+        return Promise.resolve().then(() => {
+            if (!(resource instanceof Resource)) {
```
Comment (bestbeforetoday) [inline packages/composer-runtime/lib/registry.js]: The error says it expects a `Resource` or `Concept` but the test only allows a `Resource`. Tests should cover all the allowable types.

File: packages/composer-runtime/lib/registry.js [hunk 3]
```diff
@@ -172,9 +172,17 @@ class Registry extends EventEmitter {
      * with an error.
      */
     add(resource, options) {
-        return this.accessController.check(resource, 'CREATE')
+
+        return Promise.resolve().then(() => {
+            if (!(resource instanceof Resource)) {
```
Comment (bestbeforetoday) [inline packages/composer-runtime/lib/registry.js]: The error says it expects a `Resource` or `Concept` but the test only allows a `Resource`. Tests should cover all the allowable types.

**Pull request merged**